src/lsm/compaction.zig

```diff
@@ -584,7 +622,10 @@ pub fn CompactionType(
 
             // TODO(Beat Pacing) This should really be where the compaction callback is invoked,
             // but currently that can occur multiple times per beat.
-            compaction.grid.forfeit(compaction.grid_reservation);
+            if (compaction.strategy == .table_merge) {
+                compaction.grid.forfeit(compaction.grid_reservation.?);
+                compaction.grid_reservation = null;
+            }
```

Comment:
Just to make it extra-clear that there is no reservation to forfeit:

```suggestion
            } else {
                assert(compaction.grid_reservation == null);
            }
```